src/common.rs

```diff
@@ -1,21 +1,30 @@
 //! Contains common utility & convenience functions.
 //! All functions here are intended for internal use only.
-use crate::errors::Result;
+use crate::errors::Result as CognativeResult;
 use crate::CERTIFICATES;
 use gouth::Builder;
 use std::sync::Arc;
 use tonic::{
-    metadata::MetadataValue,
+    metadata::{Ascii, MetadataValue},
+    service::Interceptor,
     transport::{Certificate, Channel, ClientTlsConfig},
+    Status,
 };
 
-/// Convenience function to return tonic Interceptor
-/// (see https://docs.rs/tonic/0.4.3/tonic/struct.Interceptor.html)
-#[allow(clippy::rc_buffer)]
-pub(crate) fn new_interceptor(token_header_val: Arc<String>) -> Result<tonic::Interceptor> {
-    let interceptor = tonic::Interceptor::new(move |mut req: tonic::Request<()>| {
-        let meta_result = MetadataValue::from_str(&token_header_val);
-
+#[derive(Clone)]
```

Comment:
I don't remember details now but interceptor is somethin like middleware in tonic right? Any reason why original implementation must be changed? Does it not work well with new tonic?

Reply:
The existing way was deprecated in 5.0.0, this was the way presented in the examples folder